File: Leetcode/JavaScript/1.js
// Leetcode 1. Two Sum

const solution = (nums, target) => {
  const N = nums.length;
  const complementMap = new Map();

  for(let i = 0; i < N; i += 1) {
    const targetSum = target - nums[i];
    if (complementMap.has(targetSum)) {
      return [complementMap.get(targetSum), i]
    }
    complementMap.set(nums[i], i);
  }
  return null;
}

console.log(solution([2, 7, 11,15], 9));
console.log(solution([3, 2, 4], 6));
console.log(solution([3, 3], 6));

File: Leetcode/Python/1.py
# Leetcode 1. Two Sum

def solution(nums, target):
  complementDict = {}
  for i, num in enumerate(nums):
    targetSum = target - num
    if targetSum in complementDict:
      return [complementDict[targetSum], i]
    complementDict[num] = i
  return None


print(solution([2, 7, 11, 15], 9))
print(solution([3, 2, 4], 6))
print(solution([3, 3], 6))